src/foundry/client/pixi/placeables/wall.d.mts

```diff
@@ -290,12 +217,171 @@ declare global {
 
     protected override _onDragLeftMove(event: PIXI.FederatedEvent): void;
 
-    // eslint-disable-next-line @typescript-eslint/no-misused-promises
-    protected override _onDragLeftDrop(event: PIXI.FederatedEvent): Promise<unknown>;
+    protected override _prepareDragLeftDropUpdates(event: PIXI.FederatedEvent): Wall.DragLeftDropUpdate[] | null;
+
+    /**
+     * @deprecated since v12, until v14
+     * @remarks "`Wall#roof` has been deprecated. There's no replacement"
+     */
+    get roof(): null;
+
+    /**
+     * @deprecated since v12, until v14
+     * @remarks "`Wall#hasActiveRoof` has been deprecated. There's no replacement"
+     */
+    get hasActiveRoof(): boolean;
+
+    /**
+     * @deprecated since v12, until v14
+     * @remarks "`Wall#identifyInteriorState` has been deprecated. It has no effect anymore and there's no replacement."
+     */
+    identifyInteriorState(): void;
+
+    /**
+     * Determine the orientation of this wall with respect to a reference point
+     * @param point - Some reference point, relative to which orientation is determined
+     * @returns An orientation in CONST.WALL_DIRECTIONS which indicates whether the Point is left, right, or collinear (both) with the Wall
+     * @deprecated since v12, until v14
+     * @remarks "`Wall#orientPoint` has been moved to {@link Edge.orientPoint | `foundry.canvas.edges.Edge#orientPoint`}"
+     */
+    orientPoint(point: Canvas.Point): CONST.WALL_DIRECTIONS;
+
+    /**
+     * Test whether to apply a configured threshold of this wall.
+     * When the proximity threshold is met, this wall is excluded as an edge in perception calculations.
+     * @param sourceType     - Sense type for the source
+     * @param sourceOrigin   - The origin or position of the source on the canvas
+     * @param externalRadius - The external radius of the source (default: `0`)
+     * @returns `true` if the wall has a threshold greater than 0 for the source type, and the source type is within that distance.
+     * @deprecated since v12, until v14
+     * @remarks "Wall#applyThreshold has been moved to {@link Edge.applyThreshold | `foundry.canvas.edges.Edge#applyThreshold`}"
+     */
+    applyThreshold(
+      sourceType: Edge.AttenuationTypes,
+      sourceOrigin: Canvas.Point,
+      externalRadius?: number | null,
+    ): boolean;
+
+    /**
+     * @deprecated since v12, until v14
+     * @remarks "`Wall#vertices` is replaced by {@link Wall.edge | `Wall#edge`}"
+     */
+    get vertices(): this["edge"];
```

Comment:
These are deprecated so I well and truly don't care to make you change a thing but I will say I'd like to phase out all uses of `this` except where strictly necessary, it's a poorly behaving type.